build: only pass --disable-dependency-tracking to packages that understand it

fanquake opened the pull request:
By blanket passing `--disable-dependency-tracking` to all depends packages we end up with warnings (i.e in `bdb` or `freetype`) like:
```bash
configure: WARNING: unrecognized options: --disable-dependency-tracking
```
Instead, only pass it to packages that actually understand it. Related to https://github.com/bitcoin/bitcoin/issues/16354.

More info on `--disable-dependency-tracking` available [here](https://www.gnu.org/software/automake/manual/html_node/Dependency-Tracking.html).

This PR also adds `--enable-option-checking` as a configure option to all applicable packages.

fanquake committed:
build: only pass --disable-dependency-tracking to packages that understand it

By blanket passing --disable-dependency-tracking to all depends packages
we end up with some warnings like:

configure: WARNING: unrecognized options: --disable-dependency-tracking

So instead, only pass it to packages that understand it.

Related to https://github.com/bitcoin/bitcoin/issues/16354.

luke-jr commented:
Why is this a problem?

laanwj commented:
Would be nice if there was a way to make it stop instead of warn on unknown configure arguments, This would make it easy to detect when say, a dependency upgrade invalidates some of the configuration.

dongcarl commented:
I think we want to make sure that given the choice, we supply `--disable-dependency-tracking`, because the default is to enable it. In the case of an upgrade, I'm more worried about upstream packages enabling the `--{dis,en}able-dependency-tracking` toggle in a version bump, and us not setting it, than just a pesky warning.

fanquake commented:
@dongcarl Would you rather a comment in `funks.mk` explaining the blanket passing rationale? I'd like to have it somewhat documented.

dongcarl commented:
> @dongcarl Would you rather a comment in `funks.mk` explaining the blanket passing rationale? I'd like to have it somewhat documented.

Let's do that!

-----

<details>
<summary><code>funks.mk</code></summary>

![giphy](https://user-images.githubusercontent.com/3445290/66057889-fb5adf80-e528-11e9-880c-6ffbd6884fa4.gif)
</details>

fanquake committed:
build: pass --enable-option-checking to applicable packages

fanquake commented:
Thanks @theuni. I've left the changes as they are, and added an additional commit that adds `--enable-option-checking` to the applicable packages. Also updated the PR description and queued up a new gitian build.

laanwj commented:
ACK 1ba49bcdc216162f513e74e371ee9f26418a390e

theuni reviewed (approved):
ACK 1ba49bcdc216162f513e74e371ee9f26418a390e
